ocaml-eglot-type-enclosing.el

```diff
@@ -0,0 +1,171 @@
+
+;;; Internal functions
+
+(defun ocaml-eglot-type-enclosing-copy ()
+  "Copy the type of the current enclosing to the Kill-ring."
+  (interactive)
+  (when ocaml-eglot-type-enclosing-current-type
+    (eglot--message "Copied `%s' to kill-ring"
+                    ocaml-eglot-type-enclosing-current-type)
+    (kill-new ocaml-eglot-type-enclosing-current-type)))
+
+(defun ocaml-eglot-type-enclosing--with-fixed-offset ()
+  "Compute the type enclosing for a dedicated offset."
+  (let* ((verbosity ocaml-eglot-type-enclosing-verbosity)
+         (index ocaml-eglot-type-enclosing-offset)
+         (at (ocaml-eglot-util--current-position-or-range))
+         (result (ocaml-eglot-req--type-enclosings at index verbosity))
+         (type (cl-getf result :type)))
+    (setq ocaml-eglot-type-enclosing-current-type type)
+    (ocaml-eglot-type-enclosing--display type t)))
+
+(defun ocaml-eglot-type-enclosing-increase-verbosity ()
+  "Increase the verbosity of the current request."
+  (interactive)
+  (if ocaml-eglot-type-enclosing-verbosity
+      (setq ocaml-eglot-type-enclosing-verbosity
+            (1+ ocaml-eglot-type-enclosing-verbosity))
+    (setq ocaml-eglot-type-enclosing-verbosity 1))
+  (ocaml-eglot-type-enclosing--with-fixed-offset))
+
+(defun ocaml-eglot-type-enclosing-decrease-verbosity ()
+  "Decrease the verbosity of the current request."
+  (interactive)
+  (if ocaml-eglot-type-enclosing-verbosity
+      (progn
+        (setq ocaml-eglot-type-enclosing-verbosity
+              (1- ocaml-eglot-type-enclosing-verbosity))
+        (when (< ocaml-eglot-type-enclosing-verbosity 1)
+          (setq ocaml-eglot-type-enclosing-verbosity nil)))
+    (setq ocaml-eglot-type-enclosing-verbosity nil))
+  (ocaml-eglot-type-enclosing--with-fixed-offset))
+
+(defun ocaml-eglot-type-enclosing-grow ()
+  "Growing of the type enclosing."
+  (interactive)
+  (when ocaml-eglot-type-enclosing-types
+    (if (>= ocaml-eglot-type-enclosing-offset
+            (1- (length ocaml-eglot-type-enclosing-types)))
+        (setq ocaml-eglot-type-enclosing-offset 0)
+      (setq ocaml-eglot-type-enclosing-offset
+            (1+ ocaml-eglot-type-enclosing-offset)))
+    (ocaml-eglot-type-enclosing--with-fixed-offset)))
+
+(defun ocaml-eglot-type-enclosing-shrink ()
+  "Display the type enclosing of a smaller enclosing if possible."
+  (interactive)
+  (when ocaml-eglot-type-enclosing-types
+    (if (<= ocaml-eglot-type-enclosing-offset 0)
+        (setq ocaml-eglot-type-enclosing-offset
+              (1- (length ocaml-eglot-type-enclosing-types)))
+      (setq ocaml-eglot-type-enclosing-offset
+            (1- ocaml-eglot-type-enclosing-offset)))
+    (ocaml-eglot-type-enclosing--with-fixed-offset)))
+
+(defun ocaml-eglot-type-enclosing--type-buffer (type-expr)
+  "Create buffer with content TYPE-EXPR of the enclosing type buffer."
+  (let ((curr-dir default-directory)
+        (current-major-mode major-mode))
```

Comment:
If this the trick for having correct syntax highlighting ? If so I think that's worth a comment :-)